service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/RegistryManager.java

```diff
@@ -59,15 +66,20 @@ public RegistryManager()
                 .httpConnectTimeout(RegistryManagerOptions.DEFAULT_HTTP_CONNECT_TIMEOUT_MS)
                 .httpReadTimeout(RegistryManagerOptions.DEFAULT_HTTP_READ_TIMEOUT_MS)
                 .build();
+        hostName = ""; // only so that hostName can be declared final
```

Comment:
Is `final` the same as `readonly` in C#?

Reply:
No, it just means that the field will always be assigned a value at constructor time. This constructor is a bit of a special case because we never intended for it to exist. Because of that, assigning an empty hostname is fine here

Reply:
isn't that what readonly means in c#?
https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/readonly

Reply:
Got it. I think the comment could be updated to explain that a bit more.

`// we don't use hostname in this ctor, so assign it an empty value to satisfy the field being marked final`